### DISCUSSION TITLE
Using any other field except 'id' to view a particular collection

### DISCUSSION THREAD

**Discussion by gaganbiswas** [pocketbase/pocketbase#3738] [Q&A] [answered]
Is there any other way to view a particular collection without using the 'id' field. 

For example:
Say, I have a website that has many products. On-click a product i am navigated to the product details page, but now i have to also pass the id along with the slug to that page via url search params or maybe something else so that the product details is fetched using the product id.

Note: The slug is unique for each product.

A possible solution to the above would be to list the products and use a filter during the request to return only the product with the particular slug. But is this a good solution?

**Comment by ganigeorgiev** [comment]
The view request (aka. `getOne`) at the moment works only with the record id.

If you need to search/retrieve a single record by a filter expression, you can use the `getFirstListItem(filter)` SDKs helper.

**Reply by gaganbiswas** [reply]
Is using the list with filter (aka. `getFirstListItem(filter)` ) has the same network usage as the `getOne`. Since I am hosting on fly.io and it has some free tier network usage so, I was wondering if using` getFirstListItem(filter)` is as efficent as `getOne`.

**Reply by ganigeorgiev** [reply, marked as answer]
Both `getFirstListItem` and `getOne` will send only 1 network request.

It depends on your filter expression, but `getOne` in general is expected to be more performant because it will search by the primary key.

**Reply by gaganbiswas** [reply]
Okay... then i guess i will stick with the getFirstListItem
